drivers/sensor/ak8975/ak8975.c

```diff
@@ -161,16 +154,23 @@ int ak8975_init(const struct device *dev)
 		return -EINVAL;
 	}
 
-	if (ak8975_read_adjustment_data(drv_data) < 0) {
+	if (ak8975_read_adjustment_data(dev) < 0) {
 		return -EIO;
 	}
 
 	return 0;
 }
 
-struct ak8975_data ak8975_data;
+static struct ak8975_data ak8975_inst_data;
+
+static const struct ak8975_config ak8975_inst_config = {
+	.i2c = I2C_DT_SPEC_INST_GET(0),
+	IF_ENABLED(DT_NODE_HAS_STATUS(DT_INST(0, invensense_mpu9150), okay),
+		   (.mpu9150_i2c = { .bus = DEVICE_DT_GET(DT_BUS(DT_INST(0, invensense_mpu9150))),
+				     .addr = DT_REG_ADDR(DT_INST(0, invensense_mpu9150)) },))
```

Comment:
using 2 i2c-dt-specs seems like adding too much complexity, because we'll always have them attached to the same bus. Here's my suggestion:

add a new DT property (phandle) to be set if sensor is embedded within an MPU9150:

```
mpu9150: mpu9150@yy {
    reg = <0xyy>;
};

ak8975: ak8975@zz {
    compatible = ...;
    reg = <0xzz>;
    embedded = <&mpu9150>;
};
```

Then, store mpu9150 address if set:
```
	.mpu9150_addr = COND_CODE_1(
		DT_INST_NODE_HAS_PROP(inst, embedded),
		(DT_REG_ADDR(DT_INST_PHANDLE(inst, embedded))), (0U)),
```

If `mpu9150_addr != 0U` (I assume 0 won't be valid?), you can activate the bypass, using the same i2c instance.

Reply:
We shouldn't do this as this binding was done to match upstream linux binding.

https://git.kernel.org/pub/scm/linux/kernel/git/devicetree/devicetree-rebasing.git/tree/Bindings/iio/gyroscope/invensense,mpu3050.yaml

Reply:
What about moving mpu9150 to its own driver and have an ak8975-pass-through property?

Reply:
I opened a PR yesterday for moving pass-through to a MPU9150 driver [#47584](https://github.com/zephyrproject-rtos/zephyr/pull/47584).

Reply:
My main issue is that we need to conform to the "linux" binding for the MPU9150 and not introduce changes to that binding.

Reply:
Is upstream linux conformance something we usually strive for or is it specific to this driver?

Reply:
its a general rule:

https://docs.zephyrproject.org/latest/build/dts/design.html#source-compatibility-with-other-operating-systems